### PR TITLE
doc(RHTAPWATCH-1049): onboard app through the UI

### PR DESCRIPTION
**Pull request by yftacherzog**
Split the onboarding process to two options.
Move some of the common tasks to separate files to keep the README from exploding.

closes: #121 
closes: #125

### CODE REVIEW

**Comment by yftacherzog** [comment]
I'm still testing this, so keeping it a draft for now.

File: docs/github-secrets.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+The process of creating a GitHub application is explained in
+[Pipelines-as-Code documentation](https://pipelinesascode.com/docs/install/github_apps/#manual-setup).
+The same secret described there, should be deployed to the `build-service` and
+`integration-service` namespaces.
+
+To do that, repeat the `kubectl create secret` command described there for the
+`pipelines-as-code` namespace also to those namespace:
+
+```bash
+kubectl -n pipelines-as-code create secret generic pipelines-as-code-secret \
```
Comment (avi-biton) [inline docs/github-secrets.md:13]: I know it is described in the PaC documentation, but to make it easy, we can add it here (the same goes for all the coomands)
```suggestion
kubectl -n pipelines-as-code create secret generic pipelines-as-code-secret \
        --from-literal github-private-key=<GH_APP_PRIVATE_KEY> \
        --from-literal github-application-id=<GH_APP_ID> \
```

Comment (yftacherzog) [reply docs/github-secrets.md:13]: If we're repeating stuff that is written there, it means we can potentially go out of sync when that changes. Here we need this command 4 times, so I'm worried about repeating it...

Anyhow, this is not a change we introduce in this PR -- we just move it to a separate file, so I prefer not to change it as part of this PR.

File: docs/github-secrets.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Configuring Github Application Secrets
+===
+
+The process of creating a GitHub application is explained in
+[Pipelines-as-Code documentation](https://pipelinesascode.com/docs/install/github_apps/#manual-setup).
+The same secret described there, should be deployed to the `build-service` and
+`integration-service` namespaces.
```
Comment (avi-biton) [inline docs/github-secrets.md]:
```suggestion
`integration-service` namespaces as well.
```

Comment (yftacherzog) [reply docs/github-secrets.md]: done

File: docs/quay.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# Configuring a Push Secret for the Build Pipeline
+
+After the build-pipeline builds an image, it will try to push it to a container registry.
+If using a registry that requires authentication, the namespace where the pipeline is
+running should be configured with a push secret for the registry.
+
+Tekton provides a way to inject push secrets into pipelines by attaching them to a
+service account.
+
+The service account used for running the pipelines is the namespace's `default` service
```
Comment (avi-biton) [inline docs/quay.md]:
```suggestion
The service account used for running the pipelines is the namespace's `appstudio-pipeline` service
```

Comment (yftacherzog) [reply docs/quay.md]: Just noticed that as well. I'll have it changed.

Comment (avi-biton) [reply docs/quay.md]: While we are on the same update, lets update this one as well:
https://github.com/konflux-ci/konflux-ci/blob/512ac0fe10a19e2c977df8e0f04fcf5fa2b4cb2f/README.md?plain=1#L580

Comment (yftacherzog) [reply docs/quay.md]: alright

**Comment by yftacherzog** [comment]
I verified the process using the UI. Next, I'll do the same without the UI.

**Comment by yftacherzog** [comment]
I verified the process with and without the UI.

**Review by avi-biton — APPROVED** [review]
lgtm

File: .github/workflows/check-toc.yaml [hunk 4]
```diff
@@ -11,11 +11,13 @@ jobs:
...
       - name: Verify ToC
         run: |
-          npx markdown-toc README.md -i
+          find . -name "*.md" | while read -r file; do
+              npx markdown-toc $file -i
+          done
           if [[ $(git status --porcelain) ]]; then
-              echo ERROR: "Did you forget to update the README table of contents?" >&2
-              echo ERROR: "Please run 'npx markdown-toc README.md -i' and update your PR" >&2
+              echo ERROR: "Did you forget to update markdowns table of contents?" >&2
```
Comment (avi-biton) [inline .github/workflows/check-toc.yaml:18]:
```suggestion
              echo ERROR: "Did you forget to update markdowns tables of contents?" >&2
```

Comment (yftacherzog) [reply .github/workflows/check-toc.yaml:18]: Was thinking about it yesterday. Not sure what is the proper plural form here.

Comment (avi-biton) [reply .github/workflows/check-toc.yaml:18]: Whatever you decide, just make sure it is consistent with https://github.com/konflux-ci/konflux-ci/blob/b05d020eb02cc8858b1eb75749c5c1196da98404/.github/workflows/check-toc.yaml#L23

**Review by gbenhaim — APPROVED** [review]

**Pull request merged**